src/app/_services/api/accounts/consent.service.ts

```diff
@@ -20,22 +19,15 @@ export class ConsentService {
         legal_last_name: string,
         student_number: string,
         date: string
-    }): Observable<string> {
-        return this
-            .http.post(this.consentAPIUrl, input, {responseType: 'text'})
-            .pipe(
-                map((response) => {
-                    if (response) {
-                        return response;
-                    }
-                },
-                (error: unknown) => {
-                    return error;
-                })
-            );
+    }): Observable<UserConsent> {
```

Comment:
Are you sure the response type is `UserConsent`?

Reply:
Yep. Upon checking the API, even when consent is declined we create a UserConsent Object like this:
```
{
    "user": "aadiraju",
    "consent": false,
    "legal_first_name": "",
    "legal_last_name": "",
    "student_number": "",
    "date": ""
}
```